src/App.tsx

```diff
@@ -1,24 +1,88 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+import { Footer } from './components/Footer';
+import { prepareTodos } from './helpers';
+import { Header } from './components/Header';
+import { ErrorType } from './types/ErrorType';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoList } from './components/TodoList';
+import { AppContext } from './AppContext';
+import { USER_ID } from './userId/userId';
 
 export const App: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    status,
+    setStatus,
+    error,
+    setError,
+  } = useContext(AppContext);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setError(ErrorType.cantLoadTodos));
+  }, [setTodos, setError]);
+
+  const filterTodos = useCallback(() => {
+    return prepareTodos({
+      todos,
+      status,
+    });
+  }, [todos, status]);
+
+  const todosOnPage = useMemo(() => filterTodos(),
```

Comment:
```suggestion
  const todosOnPage = useMemo(() => prepareTodos({
      todos,
      status,
    }),
```
`filterTodos` is redundant in this case, you can do it in `todosOnPage` and avoid creating a wrapper for `prepareTodos`